src/executor/Executor.cc

```diff
@@ -190,6 +190,11 @@ void Executor::executeParent() {
     for (auto& listener: eventListeners_) {
         listener->onPostExecute();
     }
+    // This is needed, as most earlier waits have the WNOWAIT flag.
+    if (waitpid(-1, NULL, WNOHANG) == -1 && errno != ECHILD) {
+        throw SystemException(
+                std::string("final waitpid failed: ") + strerror(errno));
+    }
```

Comment:
I think this could be ~~simplified~~ DRY-d by using [withErrnoCheck(description, ignoredErrnos, func, args...)](https://github.com/sio2project/sio2jail/blob/c1a8669b7c03c1aeda54d5757598371bafc0d150/src/common/WithErrnoCheck.h#L93-L97)

Like this:
```
withErrnoCheck("final waitpid", {ECHILD}, waitpid, -1, NULL, WNOHANG)
```
That'd be more consistent with [other code](https://github.com/sio2project/sio2jail/blob/c1a8669b7c03c1aeda54d5757598371bafc0d150/src/tracer/TraceExecutor.cc#L73-L80).

Unless you find `withErrnoCheck` confusing, then we can merge this as is.